syncd/FlexCounter.cpp

```diff
@@ -934,24 +934,21 @@ class CounterContext : public BaseCounterContext
                     auto rid = rids[i];
                     auto vid = vids[i];
                     std::vector<uint64_t> stats(counter_ids.size());
-                    if (collectData(rid, counter_ids, effective_stats_mode, false, stats)) {
-
-                        auto it_vid = m_objectIdsMap.find(vid);
-                        if (it_vid != m_objectIdsMap.end())
-                        {
-                            // Remove and re-add if vid already exists
-                            m_objectIdsMap.erase(it_vid);
-                        }
-
-                        auto counter_data = std::make_shared<CounterIds<StatType>>(rid, counter_ids);
-                        m_objectIdsMap.emplace(vid, counter_data);
-
-                        SWSS_LOG_INFO("Fallback to single call for object 0x%" PRIx64, vid);
-                    } else {
-                        SWSS_LOG_WARN("%s RID %s can't provide the statistic",  m_name.c_str(), sai_serialize_object_id(rid).c_str());
+                    if (!collectData(rid, counter_ids, effective_stats_mode, false, stats))
```

Comment:
this is not a good change. it doesn't really fix this issue. it highly depends on the SAI implementation. If SAI decide to bail out on capability check or first failure, then all / most of counters will be 0 for this port.

Reply:
if we decide to fix the flexcounter, I think we should use different stat list for different object. we did some calculations, this will cause ~2M memory increase, but it is acceptable.